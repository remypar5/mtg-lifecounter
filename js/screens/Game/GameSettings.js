import React from 'react';
import { StyleSheet, Text, Button, View, Alert, Image, Dimensions } from 'react-native';
import PropTypes from 'prop-types';

import NumberSpinner from '../../components/NumberSpinner';
import { FONT_FAMILY, COLOR_MARKED, COLOR_FOREGROUND, COLOR_BACKGROUND } from '../../utils/constants';

const { width: screenWidth, height: screenHeight } = Dimensions.get('screen');

export default class GameSettings extends React.Component {

    static lifePoints = [20, 30, 40];

    constructor(props) {
        super(props);

        this.state = {
            numberOfPlayers: 2,
            // startingLifeTotal: GameSettings.lifePoints[0]
            // numberOfPlayers: 6,
            startingLifeTotal: 888
        };
    }

    render() {
        const state = this.state;
        const { container, bgicon, label, buttonContainer, button, selected, startButton } = styles;

        return (
            <View style={ container }>
                <Image source={require('./bgicon.png')} style={[StyleSheet.absoluteFill, bgicon]} />
                <Text style={ label }>Players</Text>
                <NumberSpinner
                    min={ 1 }
                    max={ 6 }
                    stepLarge={ 1 }
                    value={ state.numberOfPlayers }
                    onChange={ (numberOfPlayers) => this.setState({numberOfPlayers}) } />

                <Text style={ label }>Lifepoints</Text>
                <View style={ buttonContainer }>
                    { GameSettings.lifePoints.map((startingLifeTotal) => (
                        <Text
                            key={`buttonLifePoints${startingLifeTotal}`}
                            style={[ button, state.startingLifeTotal === startingLifeTotal ? selected : null ]}
                            onPress={() => this.setState({ startingLifeTotal })}>
                            {startingLifeTotal}
                        </Text>
                    )) }
                </View>

                <Text
                    style={ [button, startButton] }
                    onPress={ () => this.startRound() }>
                    Start game
                </Text>
            </View>
        );
    }

    startRound() {
        const { startingLifeTotal, numberOfPlayers } = this.state;

        this.props.navigation.navigate('GameRound', { startingLifeTotal, numberOfPlayers });
    }
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: '#161616',
        flex: 1,
        flexDirection: 'column',
        alignItems: 'stretch',
        justifyContent: 'flex-end',
    },
    bgicon: {
        width: '100%',
        height: screenHeight,
    },
    label: {
        color: COLOR_FOREGROUND,
        fontSize: 25,
        textAlign: 'left',
        width: '95%',
        alignSelf: 'center'
    },
    buttonContainer: {
        height: '20%',
        flex: 1,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-around',
    },
    button: {
        flex: 1,
        height: 50,
        backgroundColor: COLOR_BACKGROUND,
        color: COLOR_FOREGROUND,
        fontFamily: FONT_FAMILY,
        fontSize: 24,
        textAlign: 'center',
        textAlignVertical: 'center',
        alignSelf: 'baseline'
    },
    startButton: {
        width: '100%',
        height: 40,
        fontSize: 36,
        borderColor: '#161616',
        borderWidth: 6,
    },
    selected: {
        backgroundColor: COLOR_MARKED
    },
});
