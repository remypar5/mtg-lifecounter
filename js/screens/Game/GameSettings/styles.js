import { StyleSheet, Dimensions } from 'react-native';

import { COLOR_MARKED, FONT_FAMILY, FONT_SIZE_SMALL, FONT_SIZE } from '../../../utils/constants';
import { normalizeFontSize } from '../../../utils';

const { height: screenHeight } = Dimensions.get('window');
const styles = StyleSheet.create({
    container: {
        flexDirection: 'column',
    },
    bgicon: {
        width: '100%',
        height: screenHeight,
    },
    buttonContainer: {
        height: '20%',
        flex: 1,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-around',
    },
    button: {
        flex: 1,
        fontFamily: FONT_FAMILY,
        paddingVertical: normalizeFontSize(3),
        fontSize: normalizeFontSize(FONT_SIZE_SMALL),
    },
    startButtonContainer: {
        width: '100%',
        padding: normalizeFontSize(2),
    },
    startButton: {
        width: '100%',
        fontSize: normalizeFontSize(FONT_SIZE),
        paddingVertical: normalizeFontSize(6),
    },
    selected: {
        backgroundColor: COLOR_MARKED,
    },
});

export default styles;
