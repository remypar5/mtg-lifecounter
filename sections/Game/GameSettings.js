import React from 'react';
import { StyleSheet, Text, Button, View, Alert } from 'react-native';
import PropTypes from 'prop-types';

import NumberSpinner from '../../components/NumberSpinner';

export default class GameSettings extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            playerAmount: 2,
            startingLifeTotal: 20
        };
    }

    render() {
        return (
            <View style={ styles.container }>
                <Text style={ styles.label }>Number of players [1-6]</Text>
                <NumberSpinner
                    min={ 1 }
                    max={ 6 }
                    value={ this.state.playerAmount }
                    onChange={ (playerAmount) => this.setState({playerAmount}) } />

                <Text style={ styles.label }>Starting lifepoints [{this.state.startingLifeTotal}]</Text>
                <View style={ styles.buttonContainer }>
                    { [20, 30, 40].map((startingLifeTotal) => (
                        <Button
                            key={`buttonLifePoints${startingLifeTotal}`}
                            style={[ styles.button, this.state.startingLifeTotal === startingLifeTotal ? styles.selected : null ]}
                            title={`${startingLifeTotal}`}
                            onPress={() => this.setState({ startingLifeTotal })} />
                    )) }
                </View>

                <Button
                    style={ styles.button }
                    title="Start game"
                    onPress={ () => this.startRound() } />
            </View>
        );
    }

    startRound() {
        const { playerAmount, startingLifeTotal } = this.state;
        
        this.props.onStart(playerAmount, startingLifeTotal);
    }
}

GameSettings.propTypes = {
    onStart: PropTypes.func.isRequired
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: 'black',
        flex: 1,
        flexDirection: 'column',
        alignItems: 'stretch'
    },
    label: {
        color: 'white',
        fontWeight: 'bold',
        fontSize: 18,
    },
    buttonContainer: {
        height: '20%',
        flex: 1,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
    },
    button: {
        height: 50,
        width: '100%',
        backgroundColor: 'steelblue',
        textAlign: 'center',
        textAlignVertical: 'center',
        color: 'white',
        alignSelf: 'baseline'
    },
    selected: {
        backgroundColor: 'red'
    },
});