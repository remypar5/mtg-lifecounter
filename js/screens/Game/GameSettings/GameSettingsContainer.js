import React from 'react';
import { StyleSheet, Image } from 'react-native';
import PropTypes from 'prop-types';

import GameSettings from './GameSettings';
import { COLOR_FOREGROUND } from '../../../utils/constants';

const styles = StyleSheet.create({
    header: {
        backgroundColor: '#101011',
    },
    headerTitle: {
        color: COLOR_FOREGROUND,
    },
});

// Ignore the next line because react-router requires containers to be actual React Components
// eslint-disable-next-line react/prefer-stateless-function
export default class GameSettingsContainer extends React.Component {
    render() {
        const { navigation } = this.props;

        return (<GameSettings navigation={navigation} />);
    }
}

GameSettingsContainer.navigationOptions = {
    title: 'Lifecounter',
    headerLeft: <Image
        style={{ width: 42, height: 42 }}
        source={{ uri: 'mipmap/ic_launcher' }}
    />,
    headerStyle: styles.header,
    headerTitleStyle: styles.headerTitle,
};

GameSettingsContainer.propTypes = {
    navigation: PropTypes.object.isRequired,
};
