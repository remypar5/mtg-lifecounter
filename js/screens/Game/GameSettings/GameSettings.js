import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';
import { NumberSpinner, Text, PageContainer, BackgroundImage } from '../../../components';
import bgiconSrc from '../bgicon.png';

export default class GameSettings extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            numberOfPlayers: 2,
            startingLifeTotal: GameSettings.lifePoints[0],
        };
    }

    startRound() {
        const { startingLifeTotal, numberOfPlayers } = this.state;

        this.props.navigation.navigate('GameRound', { startingLifeTotal, numberOfPlayers });
    }

    render() {
        const { numberOfPlayers, startingLifeTotal } = this.state;
        const {
            container, buttonContainer, button, selected, startButtonContainer, startButton,
        } = styles;

        return (
            <PageContainer style={container}>
                <BackgroundImage source={bgiconSrc} />
                <Text type="label">Players</Text>
                <NumberSpinner
                    min={1}
                    max={6}
                    size="large"
                    stepLarge={1}
                    value={numberOfPlayers}
                    onChange={(nrPlayers) => this.setState({ numberOfPlayers: nrPlayers })}
                />

                <Text type="label">Lifepoints</Text>
                <View style={buttonContainer}>
                    {GameSettings.lifePoints.map((points) => (
                        <Text
                            type="button"
                            key={`buttonLifePoints${points}`}
                            style={[button, startingLifeTotal === points ? selected : 0]}
                            onPress={() => this.setState({ startingLifeTotal: points })}
                        >
                            {points}
                        </Text>
                    ))}
                </View>

                <View style={startButtonContainer}>
                    <Text
                        type="button"
                        style={startButton}
                        onPress={() => this.startRound()}
                    >
                        Start
                    </Text>
                </View>
            </PageContainer>
        );
    }
}

GameSettings.lifePoints = [20, 30, 40];
GameSettings.propTypes = {
    navigation: PropTypes.object.isRequired,
};
