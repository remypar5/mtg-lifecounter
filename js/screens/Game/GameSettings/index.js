import GameSettings from './GameSettings';
import GameSettingsContainer from './GameSettings.container';
import styles from './styles';

export {
    GameSettings,
    GameSettingsContainer,
    styles,
};
