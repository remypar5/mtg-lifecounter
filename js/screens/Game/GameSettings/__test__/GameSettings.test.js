import React from 'react';
import { shallow } from 'enzyme';

import GameSettings from '../GameSettings';

describe('<GameSettings />', () => {
    let component;

    beforeEach(() => {
        component = shallow(<GameSettings navigation={{}} />);
    });

    it('renders without crashing', () => {
        expect(component).toMatchSnapshot();
    });
});
