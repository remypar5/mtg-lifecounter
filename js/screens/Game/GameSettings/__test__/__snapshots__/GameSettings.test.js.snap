// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameSettings /> renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GameSettings
    navigation={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Image
          source={
            Object {
              "testUri": "../../../js/screens/Game/bgicon.png",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 1334,
                "width": "100%",
              },
            ]
          }
        />,
        <Text
          style={null}
          type="label"
        >
          Players
        </Text>,
        <NumberSpinner
          max={6}
          min={1}
          onChange={[Function]}
          size="large"
          step={1}
          stepLarge={1}
          value={2}
        />,
        <Text
          style={null}
          type="label"
        >
          Lifepoints
        </Text>,
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": "20%",
              "justifyContent": "space-around",
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "flex": 1,
                  "fontFamily": "cloisterblack",
                  "fontSize": 135,
                  "paddingVertical": 34,
                },
                Object {
                  "backgroundColor": "#710000",
                },
              ]
            }
            type="button"
          >
            20
          </Text>
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "flex": 1,
                  "fontFamily": "cloisterblack",
                  "fontSize": 135,
                  "paddingVertical": 34,
                },
                null,
              ]
            }
            type="button"
          >
            30
          </Text>
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "flex": 1,
                  "fontFamily": "cloisterblack",
                  "fontSize": 135,
                  "paddingVertical": 34,
                },
                null,
              ]
            }
            type="button"
          >
            40
          </Text>
        </View>,
        <View
          style={
            Object {
              "padding": 22.5,
              "width": "100%",
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "fontSize": 180,
                "paddingVertical": 67.5,
                "width": "100%",
              }
            }
            type="button"
          >
            Start
          </Text>
        </View>,
      ],
      "style": Object {
        "alignItems": "stretch",
        "backgroundColor": "#161616",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-end",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "source": Object {
            "testUri": "../../../js/screens/Game/bgicon.png",
          },
          "style": Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 1334,
              "width": "100%",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Players",
          "style": null,
          "type": "label",
        },
        "ref": null,
        "rendered": "Players",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "max": 6,
          "min": 1,
          "onChange": [Function],
          "size": "large",
          "step": 1,
          "stepLarge": 1,
          "value": 2,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Lifepoints",
          "style": null,
          "type": "label",
        },
        "ref": null,
        "rendered": "Lifepoints",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontFamily": "cloisterblack",
                    "fontSize": 135,
                    "paddingVertical": 34,
                  },
                  Object {
                    "backgroundColor": "#710000",
                  },
                ]
              }
              type="button"
            >
              20
            </Text>,
            <Text
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontFamily": "cloisterblack",
                    "fontSize": 135,
                    "paddingVertical": 34,
                  },
                  null,
                ]
              }
              type="button"
            >
              30
            </Text>,
            <Text
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontFamily": "cloisterblack",
                    "fontSize": 135,
                    "paddingVertical": 34,
                  },
                  null,
                ]
              }
              type="button"
            >
              40
            </Text>,
          ],
          "style": Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": "20%",
            "justifyContent": "space-around",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "buttonLifePoints20",
            "nodeType": "function",
            "props": Object {
              "children": 20,
              "onPress": [Function],
              "style": Array [
                Object {
                  "flex": 1,
                  "fontFamily": "cloisterblack",
                  "fontSize": 135,
                  "paddingVertical": 34,
                },
                Object {
                  "backgroundColor": "#710000",
                },
              ],
              "type": "button",
            },
            "ref": null,
            "rendered": 20,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "buttonLifePoints30",
            "nodeType": "function",
            "props": Object {
              "children": 30,
              "onPress": [Function],
              "style": Array [
                Object {
                  "flex": 1,
                  "fontFamily": "cloisterblack",
                  "fontSize": 135,
                  "paddingVertical": 34,
                },
                null,
              ],
              "type": "button",
            },
            "ref": null,
            "rendered": 30,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "buttonLifePoints40",
            "nodeType": "function",
            "props": Object {
              "children": 40,
              "onPress": [Function],
              "style": Array [
                Object {
                  "flex": 1,
                  "fontFamily": "cloisterblack",
                  "fontSize": 135,
                  "paddingVertical": 34,
                },
                null,
              ],
              "type": "button",
            },
            "ref": null,
            "rendered": 40,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            onPress={[Function]}
            style={
              Object {
                "fontSize": 180,
                "paddingVertical": 67.5,
                "width": "100%",
              }
            }
            type="button"
          >
            Start
          </Text>,
          "style": Object {
            "padding": 22.5,
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Start",
            "onPress": [Function],
            "style": Object {
              "fontSize": 180,
              "paddingVertical": 67.5,
              "width": "100%",
            },
            "type": "button",
          },
          "ref": null,
          "rendered": "Start",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Image
            source={
              Object {
                "testUri": "../../../js/screens/Game/bgicon.png",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": 1334,
                  "width": "100%",
                },
              ]
            }
          />,
          <Text
            style={null}
            type="label"
          >
            Players
          </Text>,
          <NumberSpinner
            max={6}
            min={1}
            onChange={[Function]}
            size="large"
            step={1}
            stepLarge={1}
            value={2}
          />,
          <Text
            style={null}
            type="label"
          >
            Lifepoints
          </Text>,
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "height": "20%",
                "justifyContent": "space-around",
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontFamily": "cloisterblack",
                    "fontSize": 135,
                    "paddingVertical": 34,
                  },
                  Object {
                    "backgroundColor": "#710000",
                  },
                ]
              }
              type="button"
            >
              20
            </Text>
            <Text
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontFamily": "cloisterblack",
                    "fontSize": 135,
                    "paddingVertical": 34,
                  },
                  null,
                ]
              }
              type="button"
            >
              30
            </Text>
            <Text
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontFamily": "cloisterblack",
                    "fontSize": 135,
                    "paddingVertical": 34,
                  },
                  null,
                ]
              }
              type="button"
            >
              40
            </Text>
          </View>,
          <View
            style={
              Object {
                "padding": 22.5,
                "width": "100%",
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Object {
                  "fontSize": 180,
                  "paddingVertical": 67.5,
                  "width": "100%",
                }
              }
              type="button"
            >
              Start
            </Text>
          </View>,
        ],
        "style": Object {
          "alignItems": "stretch",
          "backgroundColor": "#161616",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-end",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": Object {
              "testUri": "../../../js/screens/Game/bgicon.png",
            },
            "style": Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 1334,
                "width": "100%",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Players",
            "style": null,
            "type": "label",
          },
          "ref": null,
          "rendered": "Players",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "max": 6,
            "min": 1,
            "onChange": [Function],
            "size": "large",
            "step": 1,
            "stepLarge": 1,
            "value": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Lifepoints",
            "style": null,
            "type": "label",
          },
          "ref": null,
          "rendered": "Lifepoints",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontFamily": "cloisterblack",
                      "fontSize": 135,
                      "paddingVertical": 34,
                    },
                    Object {
                      "backgroundColor": "#710000",
                    },
                  ]
                }
                type="button"
              >
                20
              </Text>,
              <Text
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontFamily": "cloisterblack",
                      "fontSize": 135,
                      "paddingVertical": 34,
                    },
                    null,
                  ]
                }
                type="button"
              >
                30
              </Text>,
              <Text
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontFamily": "cloisterblack",
                      "fontSize": 135,
                      "paddingVertical": 34,
                    },
                    null,
                  ]
                }
                type="button"
              >
                40
              </Text>,
            ],
            "style": Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": "20%",
              "justifyContent": "space-around",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "buttonLifePoints20",
              "nodeType": "function",
              "props": Object {
                "children": 20,
                "onPress": [Function],
                "style": Array [
                  Object {
                    "flex": 1,
                    "fontFamily": "cloisterblack",
                    "fontSize": 135,
                    "paddingVertical": 34,
                  },
                  Object {
                    "backgroundColor": "#710000",
                  },
                ],
                "type": "button",
              },
              "ref": null,
              "rendered": 20,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "buttonLifePoints30",
              "nodeType": "function",
              "props": Object {
                "children": 30,
                "onPress": [Function],
                "style": Array [
                  Object {
                    "flex": 1,
                    "fontFamily": "cloisterblack",
                    "fontSize": 135,
                    "paddingVertical": 34,
                  },
                  null,
                ],
                "type": "button",
              },
              "ref": null,
              "rendered": 30,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "buttonLifePoints40",
              "nodeType": "function",
              "props": Object {
                "children": 40,
                "onPress": [Function],
                "style": Array [
                  Object {
                    "flex": 1,
                    "fontFamily": "cloisterblack",
                    "fontSize": 135,
                    "paddingVertical": 34,
                  },
                  null,
                ],
                "type": "button",
              },
              "ref": null,
              "rendered": 40,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              onPress={[Function]}
              style={
                Object {
                  "fontSize": 180,
                  "paddingVertical": 67.5,
                  "width": "100%",
                }
              }
              type="button"
            >
              Start
            </Text>,
            "style": Object {
              "padding": 22.5,
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Start",
              "onPress": [Function],
              "style": Object {
                "fontSize": 180,
                "paddingVertical": 67.5,
                "width": "100%",
              },
              "type": "button",
            },
            "ref": null,
            "rendered": "Start",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
